addon/views/modal.js

```diff
@@ -33,6 +34,23 @@ export default Em.View.extend(
     this.set('visible', true);
   },
 
+  /* Hide with Escape */
+  turnOnEscapeEvent: function() {
+    var _this = this;
+
+    this.set('keydownListener', function(event) {
+      if (event.which == _this.escapeKeyCode) {
+        _this.get('controller').send('closeModal', _this.get('outlet'));
+      }
+    });
+
+    Em.$('body').on('keydown', this.get('keydownListener'));
```

Comment:
To solve the problem of multiple modals, did you try using the view's [`keyDown` event documented here](http://emberjs.com/api/classes/Ember.View.html#toc_event-names)?

Maybe something like:

```
export default Ember.View,extend({
  // ...

  turnOnEscapeEvent: function(event) {
    // Close modal here
  }.on('keyDown')
});
```